vllm/sampling_params.py

```diff
@@ -260,6 +261,10 @@ def __post_init__(self) -> None:
 
         self._verify_args()
         if self.use_beam_search:
+            if not envs.VLLM_ALLOW_DEPRECATED_BEAM_SEARCH:
+                raise ValueError(
+                    "Using beam search as a sampling parameter is deprecated, and will be removed in the future release. Please use the `vllm.LLM.use_beam_search` method for dedicated beam search instead, or set the environment variable `VLLM_ALLOW_DEPRECATED_BEAM_SEARCH=1` to suppress this error. For more details, see https://github.com/vllm-project/vllm/issues/8306 ."  # noqa
```

Comment:
Do we provide openAI-compatible server alternative for beam search (not sure if this is a valid need and feel free to ignore this comment)?

Reply:
no. see the future work section at https://github.com/vllm-project/vllm/pull/8726#issuecomment-2368872745